drivers/i2c/Kconfig.sf32lb

```diff
@@ -7,3 +7,13 @@ config I2C_SF32LB
 	depends on DT_HAS_SIFLI_SF32LB_I2C_ENABLED
 	help
 	  Enable I2C driver for SiFli SF32LB.
+
+if I2C_SF32LB
+
+config I2C_SF32LB_DMA
+	bool "SF32LB I2C DMA Support"
+	depends on DMA
+	help
+	  Enable DMA support for the SF32LB I2C driver.
+
+endif # I2C_SF32LB
```

Comment:
Just enable DMA if DT has dmas props set in devicetree, so no Kconfig is needed. I think uou can also optimize the driver resources by using

```
#define DMA_USED DT_ANY_INST_HAS_PROP_STATUS_OKAY(dmas)
```